docs/setup.rst

```diff
@@ -0,0 +1,90 @@
+Setting Up Corobo
+=================
+
+The following documentation illustrates the process to install Corobo and interface it with the gitter API.
+
+Installation
+------------
+
+*  To install corobo, first you need to setup a virtual environment(it is not mandatory but highly recommended). This can be done as follows:
+   ::
+     pip3 install virtualenv
+     virtualenv venv -p python
+     cd venv
+     source bin/activate ---->  For Linux
+     \Scripts\activate   ---->  For Windows
+*  Next, you can clone the corobo package via git and install the requirements(dependent packages).
+   ::
+     git clone https://github.com/coala/corobo.git
+     pip install -r requirements.txt
+
+Configuration of Environment variables(In text mode)
+--------------------------------------------------
+
+*  The following environment variables are required to setup corobo in text mode:
+   ::
+     export BOT_PREFIX="any_name"
+     export BACKEND="Text"
+     export BOT_USERNAME="@any_other_name"
+   (**Note:** The BOT_USERNAME always starts with a @ for a person and # for a room)
+
+Testing Corobo
+---------------------------
+
+*  To run the bot use:
+   ::
+     errbot
+*  You can test the bot using the following commands:
+   ::
+     bot_prefix help    ----> Lists the set of commands available with the bot
+     bot_prefix history ----> Lists the command history
+     
+     
+Setting up Corobo with gitter backend
+=====================================
+
+#. Clone the err-backend-gitter repository in your "corobo" folder.
+   ::
+     git clone https://github.com/errbotio/err-backend-gitter.git
+     
+#. Configure the environment variables(for gitter-backend).
+   ::
+     export BACKEND="Gitter"
+     export BOT_EXTRA_BACKEND_DIR="/path_to/err-backend-gitter"
+#. To make the bot interact with the gitter API you will need a github account(this account can be used to login to gitter) to generate the gitter access token.
+   **Remember it is recommended to use a separate account(other than the admin account) for the bot.** 
+   Once the account is setup, run the following script.
+   ::
+     ./oauth.py
+   This will guide you with the gitter token generation process.
```

Comment:
What?

Reply:
On running this file it guides us with the process generating the gitter bit token.

Reply:
Can you link this file?

Reply:
This is the file.. https://github.com/errbotio/err-backend-gitter/blob/14a428bdd0597b473605264c822855da979bd916/oauth.py
This point is also mentioned in the gitter backend documentation.

Reply:
This looks very redundant, and you didn't tell them that this file belongs to the gitter backend directory. I don't think we need this as it is not even a part of corobo.